app/isr/[category]/page.tsx

```diff
@@ -1,8 +1,91 @@
-// Incremental Static Regeneration (ISR)
+import { Props, Article } from '@/app/types';
+import ArticleCard from '@/components/ArticleCard/ArticleCard';
+import { Button } from '@/components/ui/button';
+import Link from 'next/link';
 
+// Incremental Static Regeneration (ISR)
 // Use https://newsapi.org API to fetch news data by category.
 // This API will require you to create an account to get an API key.
 
-export default async function NewsPage() {
-  return <div>ISR</div>;
+const CATEGORIES = [
+  'business',
+  'entertainment',
+  'general',
+  'health',
+  'science',
+  'sports',
+  'technology',
+];
+
+const getNewsByCategory = async (category: string) => {
+  console.log(
+    `Fetching news for category: ${category} at ${new Date().toISOString()}`
+  );
+
+  try {
+    const response = await fetch(
+      `${process.env.ISR_BASE_URL}?country=us&category=${category}&apiKey=${process.env.ISR_NEWS_API_KEY}`
```

Comment:
@ironyhouse URL is wrong, some paths are missing. This causing response.json() below to throw an error

Reply:
I've configured ENV wrongly. That was an issue